File: Problem_Solving_CPP/Learning/AS/03/01_RemoveDuplicateFromSortedArray.cpp
#include<iostream>
using namespace std;

int remDupli(vector<int> &arr){
    int j = 0;
    for (int i = 0 ; i<arr.size() ; i++){
        if(arr[i]!=arr[j]){
            j++;
            swap(arr[i], arr[j]);
        }
    }
    for (int i = 0 ; i<=j ; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
    return j+1; 
}

int main (){
    vector<int>arr = {0,1,1,1,2,3,3,4,4,5};
    cout<<remDupli(arr)<<" unique element in given array";
}

File: Problem_Solving_CPP/Learning/AS/03/01_RemoveDuplicateFromSortedArray.js
let arr = [0,1,1,1,2,2,3,3,4];

function dupli(arr){
    let pointer = 0;
    for (let  i = 0; i< arr.length ; i++){
        if (arr[i]!= arr[pointer]){
            pointer++;
            arr[pointer]= arr[i]
        }
    }
    return pointer+1;
}
console.log( dupli(arr));